spinnaker/spinnaker_system/google_http_upsert_server_test.py

```diff
@@ -0,0 +1,195 @@
+                                  'insert',
+                                  resource_type='sslCertificates',
+                                  project=bindings['GCE_PROJECT'],
+                                  body=ssl_cert)
+    ssl_cert['name'] = cls.SECOND_CERT
+    compute_agent.invoke_resource(context,
+                                  'insert',
+                                  resource_type='sslCertificates',
+                                  project=bindings['GCE_PROJECT'],
+                                  body=ssl_cert)
+
+
+  @classmethod
+  def tearDownClass(cls):
+    runner = citest.base.TestRunner.global_runner()
+    bindings = runner.bindings
+    context = ExecutionContext()
+    compute_agent = gcp.GcpComputeAgent.make_agent(scopes=SCOPES,
+                                                   credentials_path=bindings['GCE_CREDENTIALS_PATH'])
+    compute_agent.invoke_resource(context,
+                                  'delete',
+                                  resource_type='sslCertificates',
+                                  project=bindings['GCE_PROJECT'],
+                                  sslCertificate=cls.FIRST_CERT)
+    compute_agent.invoke_resource(context,
+                                  'delete',
+                                  resource_type='sslCertificates',
+                                  project=bindings['GCE_PROJECT'],
+                                  sslCertificate=cls.SECOND_CERT)
+
+
+  @property
+  def scenario(self):
+    return citest.base.TestRunner.global_runner().get_shared_data(
+        upsert.GoogleHttpLoadBalancerTestScenario)
+
+  def test_a_upsert_lb(self):
+    self.run_test_case(self.scenario.upsert_full_load_balancer())
+
+  def test_b_delete_lb(self):
+    self.run_test_case(self.scenario.delete_load_balancer())
+
+  def test_c_upsert_min_lb(self):
+    self.run_test_case(self.scenario.upsert_min_load_balancer())
+
+  def test_d_upsert_full_lb(self):
+    self.run_test_case(self.scenario.upsert_full_load_balancer())
+
+  def test_e_add_security_group(self):
+    self.run_test_case(self.scenario.add_security_group())
+
+  def test_f_add_server_group(self):
+    self.run_test_case(self.scenario.add_server_group())
+
+  def test_g_change_hc(self):
+    self.run_test_case(self.scenario.change_health_check())
+
+  def test_h_change_bs(self):
+    self.run_test_case(self.scenario.change_backend_service())
+
+  def test_i_add_host_rule(self):
+    self.run_test_case(self.scenario.add_host_rule())
+
+  def test_j_update_host_rule(self):
+    self.run_test_case(self.scenario.update_host_rule())
+
+  def test_k_update_port_range(self):
+    self.run_test_case(self.scenario.update_port_range())
+
+  def test_l_add_cert(self):
+    self.run_test_case(self.scenario.add_cert(self.__class__.FIRST_CERT))
+
+  def test_m_change_cert(self):
+    self.run_test_case(self.scenario.change_cert(self.__class__.SECOND_CERT))
+
+  def test_n_delete_server_group(self):
```

Comment:
FWIW, I'd been naming delete operations from the back to the front so that future inserts/removes would be easier.

Really I'd like to be able to specify that the test suite built from this leaves the order in declaration order rather than doing a lexical sort. I was never able to get that working. If you know how to do/fix that, then that would be great.